tsconfig.json

```diff
@@ -1,69 +1,63 @@
-    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
+        /* Module Resolution Options */
+        "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
+        "baseUrl": ".",
+        "paths": {
+            "*": ["src/types/*", "node_modules/*"]
+        },
+        // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
+        // "typeRoots": [],                       /* List of folders to include type definitions from. */
+        // "types": [],                           /* Type declaration files to be included in compilation. */
+        // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+        "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
+        // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
 
-    /* Source Map Options */
-    // "sourceRoot": "./",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
-    // "mapRoot": "./",                       /* Specify the location where debugger should locate map files instead of generated locations. */
-    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
-    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
+        /* Source Map Options */
+        // "sourceRoot": "./",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
+        // "mapRoot": "./",                       /* Specify the location where debugger should locate map files instead of generated locations. */
+        // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
+        // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
 
-    /* Experimental Options */
-    // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
-    // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
-    "resolveJsonModule": true,
-  },
-  "exclude": [
-    "src/__test__/**/*",
```

Comment:
`src/__test__/` is reserved for test vectors. We have intentionally excluded it all from the tsconfig for a reason.

Reply:
I haven't removed it. It's at the bottom, but I just excluded `src/__test__/compile`. (Would have not built without it). Is there a reason to exclude it all i.e. src/__test__/**/* instead of src/__test__/compile ?

Reply:
Yes: *all* the subfolders under `src/__test__/` are test vectors, not just `src/__test__/compile`. The files under `src/__test__/compile` are test vectors for the `src/compile.test.ts` suite, and there may be another test suite in the future which also requires test vector files with a `.ts` extension.